### ISSUE TITLE
Removing Sheet Errors With Vertex for address missing in AddressMapping

### ISSUE THREAD

**Issue by BrianHung** [handsontable/hyperformula#1121]
### Description

Running into a bug where when removing a sheet, it runs into a self-circular dependency bug trying to delete cell values that reference itself. Workaround has been to clear sheet before removing.

### Demo
https://codesandbox.io/s/small-thunder-4j9w0m?file=/src/hyperformulaConfig.js

**Comment by sequba** [comment]
@BrianHung, thank you for reporting this issue. The demo you posted does not show this behavior. You might have provided wrong link by mistake?

**Comment by BrianHung** [comment]
Oops, it must've not saved: https://codesandbox.io/s/billowing-monad-obgxg9?file=/src/hyperformulaConfig.js


```ts
import HyperFormula from "hyperformula";

// Create an empty HyperFormula instance.
const hf = HyperFormula.buildEmpty({
  precisionRounding: 10,
  licenseKey: "gpl-v3"
});

// Add a new sheet and get its id.
const table1 = hf.addSheet("table1");
const sheet1 = hf.getSheetId(table1);

// Fill the HyperFormula sheet with data.
hf.setCellContents(
  {
    row: 0,
    col: 0,
    sheet: sheet1
  },
  [["1/5/22", "4/13/22", "=B1-A1"]]
);

console.log(sheet1);

// hf.clearSheet(sheet1);
hf.removeSheet(sheet1);
```

@sequba

**Comment by sequba** [comment]
@BrianHung ,
 
I'm more than happy to announce that we just released [HyperFormula 2.3.1](https://github.com/handsontable/hyperformula/releases/tag/2.3.1) where this issue is fixed. We are closing this issue as solved. If there is anything that won't work for you after updating, please leave a comment.
